src/common/Smi.Common/Messaging/ControlMessageConsumer.cs

```diff
@@ -59,116 +59,108 @@ public class ControlMessageConsumer : Consumer
             StopHost += () => stopEvent("Control message stop");
         }
 
-
         /// <summary>
         /// Recreate ProcessMessage to specifically handle control messages which wont have headers,
         /// and shouldn't be included in any Ack/Nack counts
         /// </summary>
-        /// <param name="model"></param>
         /// <param name="e"></param>
         public override void ProcessMessage(BasicDeliverEventArgs e)
         {
             try
             {
-                ProcessMessageImpl(null, e);
-            }
-            catch (Exception exception)
-            {
-                Fatal("ProcessMessageImpl threw unhandled exception", exception);
-            }
-        }
+                // For now we only deal with the simple case of "smi.control.<who>.<what>". Can expand later on depending on our needs
+                // Queues will be deleted when the connection is closed so don't need to worry about messages being leftover
 
-        /// <summary>
-        /// Ensures the control queue is cleaned up on exit. Should have been deleted already, but this ensures it
-        /// </summary>
-        public override void Shutdown()
-        {
-            using (IConnection connection = _factory.CreateConnection(_processName + _processId + "-ControlQueueShutdown"))
-            using (IModel model = connection.CreateModel())
-            {
-                Logger.Debug("Deleting control queue: " + ControlConsumerOptions.QueueName);
-                model.QueueDelete(ControlConsumerOptions.QueueName);
-            }
-        }
+                Logger.Info("Control message received with routing key: " + e.RoutingKey);
 
-        protected override void ProcessMessageImpl(IMessageHeader header, BasicDeliverEventArgs e)
-        {
-            // For now we only deal with the simple case of "smi.control.<who>.<what>". Can expand later on depending on our needs
-            // Queues will be deleted when the connection is closed so don't need to worry about messages being leftover
+                string[] split = e.RoutingKey.ToLower().Split('.');
+                string body = GetBodyFromArgs(e);
 
-            Logger.Info("Control message received with routing key: " + e.RoutingKey);
+                if (split.Length < 4)
```

Comment:
Would be nice if this was a method or class we could unit test e.g. ControlMessageParser

Reply:
Agree, but that was the existing behavior.